src/Evolution/Systems/GrMhd/ValenciaDivClean/Initialize.hpp

```diff
@@ -66,14 +80,34 @@ struct Initialize {
           db::get<::Tags::Coordinates<Dim, Frame::Inertial>>(box);
 
       // Set initial data from analytic solution
-      using solution_tag = OptionTags::AnalyticSolutionBase;
       PrimitiveVars primitive_vars{num_grid_points};
-      primitive_vars.assign_subset(Parallel::get<solution_tag>(cache).variables(
-          inertial_coords, initial_time,
-          typename Metavariables::analytic_variables_tags{}));
-
-      auto equation_of_state =
-          Parallel::get<solution_tag>(cache).equation_of_state();
+      auto equation_of_state = make_overloader(
+          [ initial_time, &inertial_coords ](
+              std::true_type /*is_analytic_solution*/, const auto prim_vars,
+              const auto& local_cache) noexcept {
+            using solution_tag = OptionTags::AnalyticSolutionBase;
+            prim_vars->assign_subset(
+                Parallel::get<solution_tag>(local_cache)
+                    .variables(
+                        inertial_coords, initial_time,
+                        typename Metavariables::analytic_variables_tags{}));
+            return Parallel::get<solution_tag>(local_cache).equation_of_state();
+          },
+          [&inertial_coords](std::false_type /*is_analytic_solution*/,
+                             const auto prim_vars,
+                             const auto& local_cache) noexcept {
+            using analytic_data_tag = OptionTags::AnalyticDataBase;
+            prim_vars->assign_subset(
+                Parallel::get<analytic_data_tag>(local_cache)
+                    .variables(
+                        inertial_coords,
+                        typename Metavariables::analytic_variables_tags{}));
+            return Parallel::get<analytic_data_tag>(local_cache)
+                .equation_of_state();
+          })(
+          std::integral_constant<
```

Comment:
the first argument can be`detail::has_analytic_solution_alias<Metavariables>`, which is either `true_type` or `false_type` (and below). It seems these are the only places the `_v` version is used, so that can probably be omitted as well.